src/sandstorm/web-publishing.capnp

```diff
@@ -0,0 +1,81 @@
+# Sandstorm - Personal Cloud Sandbox
+# Copyright (c) 2015 Sandstorm Development Group, Inc. and contributors
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@0xd5d3e63bd0a552b6;
+
+$import "/capnp/c++.capnp".namespace("sandstorm");
+
+using Util = import "util.capnp";
+using Handle = Util.Handle;
+
+interface WebSite {
+  # Represents an HTTP URL to which a web site can be published.
+  #
+  # Make a powerbox request for `WebSite` when you wish to publish a web site accessible over HTTP
+  # without going through the Sandstorm UI.
+
+  getUrl @0 () -> (path :Text);
+  # Get the URL of this resource.
+
+  struct Entity {
```

Comment:
This interface should probably grow support for permitting cross-origin queries, so we can remove the hacks at https://github.com/sandstorm-io/sandstorm/blob/cf984260cca26a1d5b7abad82ee8fed73e5ff2d9/shell/server/pre-meteor.js#L86..L97